CHANGELOG.md

```diff
@@ -1,5 +1,13 @@
 # Changelog
 
+### 1.30.0 (TBD)
+* [JENKINS-55353](https://issues.jenkins-ci.org/browse/JENKINS-55353) Upgrade Jenkins core to 2.150.1
+* [JENKINS-5149](https://issues.jenkins-ci.org/browse/JENKINS-5149) Overwrite remoting.jar only when necessary.
+
+#### Breaking changes
+* this version remove deprecated methods on SSHLauncher see [JENKINS-55353](https://issues.jenkins-ci.org/browse/JENKINS-55353)
```

Comment:
The issue does not explain why this was done at all.

Reply:
This was a long time ago, about 4 (or more) years, I deprecated a bunch of constructors, after a time enough to make pluggins the change, IIRC a year or so, those deprecated constructors were removed. The reason to remove those constructors is because there were many and it was a mess to have to maintain 5-10 constructors deprecated every time you change something in a filed, also maintain test to check that those Contructor s still works, so for maintenance reasons I have decided to remove them.

Reply:
Also, constructors with ten or more parameters where only three are mandatory does not have any sense

Reply:
I'm not arguing there's no possible reason for the change, I'm pointing out the supposed justification is missing. At least that's my reading of a changelog entry that points at a Jira issue.

And the explanation about warnings only help with production code (which BTW is generally not something I'd recommend, warnings are seen only by admins, potentially extremely frequently, while not something they can do anything about other than report bugs). Modernizing `groovy` plugin and adapting its tests can still happen years later.

Reply:
Yep, maybe a warning in the logs was not the best way to a ounce the change, but it was in the release notes too, and when the change become a breaking change it was in the release notes as breaking change, the change was four years ago, I do not remember the details but IIRC I search for plugins using those constructors and made some PRs, but I really donor remember it, it was four year ago